scripting/get5/kniferounds.sp

```diff
@@ -66,56 +66,127 @@ static bool AwaitingKnifeDecision(int client) {
+          HandleVotes();
+        }
+    } else if(g_bVoteStart && !g_bPlayerCanVote[client]) {
+      Get5_Message(client, "%t", "VoteAlreadyCast");
+    } else {
+      return Plugin_Stop;
+    }
   }
+
   return Plugin_Handled;
 }
 
-public Action Command_Swap(int client, int args) {
+// If everyone on the winning team has already voted
+// Call HandleVotes();
+public Action Command_VoteT(int client, int args) {
   if (AwaitingKnifeDecision(client)) {
-    EndKnifeRound(true);
-    Get5_MessageToAll("%t", "TeamDecidedToSwapInfoMessage",
-                      g_FormattedTeamNames[g_KnifeWinnerTeam]);
-  } else if (g_GameState == Get5State_Warmup && g_InScrimMode &&
-             GetClientMatchTeam(client) == MatchTeam_Team1) {
-    PerformSideSwap(true);
-  }
-  return Plugin_Handled;
-}
+    if (g_bVoteStart && g_bPlayerCanVote[client]) {
+      g_bPlayerCanVote[client] = false;
+      g_iVoteTs++;
+      Get5_MessageToTeam(g_KnifeWinnerTeam, "%t", "VoteTCast");
+
+      bool runFinal = true;
+      for (int i = 1; i <= MaxClients; i++) {
+        if (AwaitingKnifeDecision(i) && g_bPlayerCanVote[i]) {
+          runFinal = false;
+        }
+      }
 
-public Action Command_Ct(int client, int args) {
-  if (IsPlayer(client)) {
-    if (GetClientTeam(client) == CS_TEAM_CT)
-      FakeClientCommand(client, "sm_stay");
-    else if (GetClientTeam(client) == CS_TEAM_T)
-      FakeClientCommand(client, "sm_swap");
+      if (runFinal) {
+        HandleVotes();
+      }
+    } else if(g_bVoteStart && !g_bPlayerCanVote[client]) {
+      Get5_Message(client, "%t", "VoteAlreadyCast");
+    } else {
+      return Plugin_Stop;
+    }
   }
-
-  LogDebug("cs team = %d", GetClientTeam(client));
-  LogDebug("m_iCoachingTeam = %d", GetEntProp(client, Prop_Send, "m_iCoachingTeam"));
-  LogDebug("m_iPendingTeamNum = %d", GetEntProp(client, Prop_Send, "m_iPendingTeamNum"));
-
+  
   return Plugin_Handled;
 }
 
-public Action Command_T(int client, int args) {
-  if (IsPlayer(client)) {
-    if (GetClientTeam(client) == CS_TEAM_T)
-      FakeClientCommand(client, "sm_stay");
-    else if (GetClientTeam(client) == CS_TEAM_CT)
-      FakeClientCommand(client, "sm_swap");
-  }
-  return Plugin_Handled;
-}
+// public Action Command_Stay(int client, int args) {
+//   if (AwaitingKnifeDecision(client)) {
+//     EndKnifeRound(false);
+//     Get5_MessageToAll("%t", "TeamDecidedToStayInfoMessage",
+//                       g_FormattedTeamNames[g_KnifeWinnerTeam]);
+//   }
+//   return Plugin_Handled;
+// }
+
+// public Action Command_Swap(int client, int args) {
+//   if (AwaitingKnifeDecision(client)) {
+//     EndKnifeRound(true);
+//     Get5_MessageToAll("%t", "TeamDecidedToSwapInfoMessage",
+//                       g_FormattedTeamNames[g_KnifeWinnerTeam]);
+//   } else if (g_GameState == Get5State_Warmup && g_InScrimMode &&
+//              GetClientMatchTeam(client) == MatchTeam_Team1) {
+//     PerformSideSwap(true);
+//   }
+//   return Plugin_Handled;
+// }
```

Comment:
Big block of redundant comments

Reply:
See 2nd conversation in regards to this.